client/jetpack-connect/test/plans.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @format
+ * @jest-environment jsdom
+ */
+/**
+ * External dependencies
+ */
+import { mount, render } from 'enzyme';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import PlansWrapper, { getSitePlans, SELECTED_SITE, SITE_PLAN_PRO } from './lib/plans';
+import { PLAN_JETPACK_BUSINESS } from 'lib/plans/constants';
+
+jest.mock( 'components/data/query-plans', () => require( 'components/empty-component' ) );
+jest.mock( 'components/data/query-site-plans', () => require( 'components/empty-component' ) );
+jest.mock( 'jetpack-connect/happychat-button', () => require( 'components/empty-component' ) );
+jest.mock( 'my-sites/plan-features', () => require( 'components/empty-component' ) );
+
+describe( 'Plans', () => {
+	test( 'should render with no plan (free)', () => {
+		const wrapper = render( <PlansWrapper /> );
+
+		expect( wrapper ).toMatchSnapshot();
+	} );
+
+	test( 'should render with a paid plan', () => {
+		const wrapper = render(
+			<PlansWrapper
+				selectedSite={ { ...SELECTED_SITE, plan: SITE_PLAN_PRO } }
+				sitePlans={ getSitePlans( PLAN_JETPACK_BUSINESS ) }
+			/>
+		);
+
+		expect( wrapper ).toMatchSnapshot();
+	} );
+
+	test( 'should redirect if Atomic', () => {
+		const goBackToWpAdmin = jest.fn();
+
+		const atomicPlans = (
+			<PlansWrapper
+				goBackToWpAdmin={ goBackToWpAdmin }
+				isAutomatedTransfer={ true }
+				selectedSite={ {
+					...SELECTED_SITE,
+					plan: SITE_PLAN_PRO,
+					is_automated_transfer: true,
+				} }
+				sitePlans={ getSitePlans( PLAN_JETPACK_BUSINESS ) }
+			/>
+		);
+
+		const wrapper = render( atomicPlans );
+
+		expect( wrapper ).toMatchSnapshot();
```

Comment:
I do wonder if this `render` test has much value, since there are Atomic tests in mount and update lifecycle methods:

https://github.com/Automattic/wp-calypso/blob/33c6425d634bca537e2e0bc0b57bef9bb1fe3137/client/jetpack-connect/plans.jsx#L69-L85

Reply:
Yeah I think really we are interested in the redirect, not the render.